special-pages/pages/new-tab/app/favorites/components/Favorites.js

```diff
@@ -1,21 +1,28 @@
-import { h } from 'preact';
-import { useId, useMemo } from 'preact/hooks';
+import { Fragment, h } from 'preact';
+import { useContext, useEffect, useId, useLayoutEffect, useMemo, useRef, useState } from 'preact/hooks';
 import { memo } from 'preact/compat';
 import cn from 'classnames';
 
 import styles from './Favorites.module.css';
-import { Placeholder, PlusIconMemo, TileMemo } from './Tile.js';
 import { ShowHideButton } from '../../components/ShowHideButton.jsx';
 import { useTypedTranslationWith } from '../../types.js';
 import { usePlatformName } from '../../settings.provider.js';
 import { useDropzoneSafeArea } from '../../dropzone.js';
+import { TileRow } from './TileRow.js';
+import { FavoritesContext } from './FavoritesProvider.js';
 
 /**
  * @typedef {import('../../../../../types/new-tab.js').Expansion} Expansion
  * @typedef {import('../../../../../types/new-tab.js').Favorite} Favorite
  * @typedef {import('../../../../../types/new-tab.js').FavoritesOpenAction['target']} OpenTarget
  */
 export const FavoritesMemo = memo(Favorites);
+export const ROW_CAPACITY = 6;
+/**
+ * Note: These values MUST match exactly what's defined in the CSS.
+ */
+const ITEM_HEIGHT = 98;
+const ROW_GAP = 8;
```

Comment:
Any way to tie them together? For me it was not obvious which property in C-S-S? Maybe the comment can mention it.

Reply:
Yeah we could technically render an item 'offset' so to speak, and measure it. good idea for a follow up for sure.